pkg/cache/server/config.go

```diff
@@ -36,13 +59,92 @@ type CompletedConfig struct {
 // Complete fills in any fields not set that are required to have valid data. It's mutating the receiver.
 func (c *Config) Complete() (CompletedConfig, error) {
 	return CompletedConfig{&completedConfig{
-		Options: c.Options,
+		Options:       c.Options,
+		ApiExtensions: c.ApiExtensions.Complete(),
+		EmbeddedEtcd:  c.EmbeddedEtcd.Complete(),
 	}}, nil
 }
 
 func NewConfig(opts *cacheserveroptions.CompletedOptions) (*Config, error) {
 	c := &Config{
 		Options: opts,
 	}
+	if opts.EmbeddedEtcd.Enabled {
+		var err error
+		c.EmbeddedEtcd, err = embeddedetcd.NewConfig(opts.EmbeddedEtcd, opts.Etcd.EnableWatchCache)
+		if err != nil {
+			return nil, err
+		}
+	}
+	serverConfig := genericapiserver.NewRecommendedConfig(apiextensionsapiserver.Codecs)
+
+	if err := opts.ServerRunOptions.ApplyTo(&serverConfig.Config); err != nil {
+		return nil, err
+	}
+	if err := opts.Etcd.ApplyTo(&serverConfig.Config); err != nil {
+		return nil, err
+	}
+	if err := opts.SecureServing.ApplyTo(&serverConfig.Config.SecureServing, &serverConfig.Config.LoopbackClientConfig); err != nil {
+		return nil, err
+	}
+	if err := opts.Authentication.ApplyTo(&serverConfig.Config.Authentication, serverConfig.SecureServing, serverConfig.OpenAPIConfig); err != nil {
+		return nil, err
+	}
+	if err := opts.Authorization.ApplyTo(&serverConfig.Config.Authorization); err != nil {
+		return nil, err
+	}
+
+	if err := opts.APIEnablement.ApplyTo(&serverConfig.Config, apiextensionsapiserver.DefaultAPIResourceConfigSource(), apiextensionsapiserver.Scheme); err != nil {
```

Comment:
Did we make a determination about whether or not the cache would need to understand storage versions & be doing something intelligent with the scheme for serialization, as opposed to being "dumb" and serving whatever?

Reply:
it should be "dumb", I recall wiring the ApiEnablement because it didn't want to start.

I think that for build-in types we would have to wire some conversion functions and webhooks for CRDs.